Makefile

```diff
@@ -1,33 +1,58 @@
+# This Makefile serves a few purposes:
+#
+# 1. It provides an interface to iterate quickly while developing the C++ code in src/
+# by typing `make` or `make debug`. To make iteration as fast as possible it calls out
+# directly to underlying build tools and skips running steps that appear to have already
+# been run (determined by the presence of a known file or directory). What `make` does is
+# the same as running `npm install --build-from-source` except that it is faster when
+# run a second time and may break if your build deps failed to install but their directories
+# do exist. If you hit a build error do to this, you should be able to fix it by running
+# `make distclean` and then `make` again.
+#
+# 2. It provides a few commands that call out to external scripts like `make coverage` or
+# `make tidy`. These scripts can be called directly but this Makefile provides a more uniform
+# interface to call them.
+#
+# To learn more about the build system see https://github.com/mapbox/node-cpp-skel/blob/master/docs/extended-tour.md#builds
+
 MODULE_NAME := $(shell node -e "console.log(require('./package.json').binary.module_name)")
 
 # Whether to turn compiler warnings into errors
 export WERROR ?= true
 
 default: release
 
-node_modules:
-	# install deps but for now ignore our own install script
-	# so that we can run it directly in either debug or release
+# install deps but for now ignore our own install script
+# so that we can run it directly in either debug or release
+node_modules/nan:
 	npm install --ignore-scripts
 
-release: node_modules
+mason_packages/headers: node_modules/nan
+	node_modules/.bin/mason-js install
+
+mason_packages/.link/include: mason_packages/headers
+	node_modules/.bin/mason-js link
+
+build-deps: mason_packages/.link/include
```

Comment:
Perhaps we can add a code comment here to describe what `build-deps` is since it is now involved in many of the other make commands.

Reply:
Started on this in https://github.com/mapbox/node-cpp-skel/pull/111/commits/6979edbf1571433d21880e30f297b3ab281913f6